sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -501,9 +496,16 @@ class CudfRegexTranspiler(replace: Boolean) {
           // falling back to CPU
           throw new RegexUnsupportedException(nothingToRepeat)
         }
-        if (replace && parts.length == 1 && (isRegexChar(parts.head, '^')
-            || isRegexChar(parts.head, '$'))) {
-          throw new RegexUnsupportedException("regexp_replace on GPU does not support ^ or $")
+        if (parts.forall {
+          case RegexChar(ch) => ch == '^' || ch == '$'
```

Comment:
Do we need to worry about any degenerate cases of other directives (e.g.: empty group, character class, etc.) that would essentially be equivalent to just having these characters in the pattern?

Reply:
For empty groups, we would fall back to CPU, but I did find one case that is not detected by this rule (`(^)($)`) so will spend some more time on this.